indico/modules/rb/client/js/common/rooms/RoomEditModal.jsx

```diff
@@ -8,951 +8,309 @@
-    errors.number = Translate.string('Please provide a number.');
-  }
-  if (capacity < 1) {
-    errors.capacity = Translate.string('Please provide a valid capacity number.');
-  }
-  if (surfaceArea && surfaceArea < 1) {
-    errors.surfaceArea = Translate.string('Please provide a valid surface area.');
-  }
-  if (maxAdvanceDays !== null && maxAdvanceDays < 1) {
-    errors.maxAdvanceDays = Translate.string(
-      'The max. advance booking time must be at least 1 day or empty.'
-    );
-  }
-  if (bookingLimitDays !== null && bookingLimitDays < 1) {
-    errors.bookingLimitDays = Translate.string(
-      'The max. booking duration must be at least 1 day or empty.'
-    );
-  }
-  const notificationPeriodFields = [
-    'notificationBeforeDays',
-    'notificationBeforeDaysWeekly',
-    'notificationBeforeDaysMonthly',
-    'notificationBeforeEndDaily',
-    'notificationBeforeEndWeekly',
-    'notificationBeforeEndMonthly',
-  ];
-
-  notificationPeriodFields.forEach(field => {
-    const value = fields[field];
-    if (isInvalidNotificationPeriod(value)) {
-      errors[field] = Translate.string('Number of days must be between 1 and 30');
-    }
+import {getAllEquipmentTypes} from './selectors';
+
+function RoomEditModal({roomId, locationId, onClose, afterCreation}) {
+  const favoriteUsersController = useFavoriteUsers();
+  const [permissionManager, permissionInfo] = usePermissionInfo();
+  const equipmentTypes = useSelector(getAllEquipmentTypes);
+  const dispatch = useDispatch();
+
+  const [globalAttributes, setGlobalAttributes] = useState([]);
+  const [newRoomId, setNewRoomId] = useState(null);
+  const [wasEverUpdated, setWasEverUpdated] = useState(null);
+  const [tabsWithError, setTabsWithError] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  const [roomDetails, setRoomDetails] = useState({
+    acl_entries: [],
+    available_equipment: [],
+    notification_emails: [],
+    notifications_enabled: true,
+    end_notifications_enabled: true,
+    is_reservable: true,
+    reservations_need_confirmation: false,
+    protection_mode: 'public',
+  });
+  const [roomAttributes, setRoomAttributes] = useState([]);
+  const [roomAvailability, setRoomAvailability] = useState({
+    bookable_hours: [],
+    nonbookable_periods: [],
   });
 
-  if (nonbookablePeriods && nonbookablePeriods.some(x => !x.startDt || !x.endDt)) {
-    errors.nonbookablePeriods = Translate.string('Please provide valid non-bookable periods.');
-  }
-  if (!owner) {
-    errors.owner = Translate.string('You need to specify the owner of the room.');
-  }
-  if (latitude !== null && typeof latitude !== 'number') {
-    errors.latitude = Translate.string('Coordinates should be valid numbers.');
-  }
-  if (longitude !== null && typeof longitude !== 'number') {
-    errors.longitude = Translate.string('Coordinates should be valid numbers.');
-  }
-  return errors;
-}
+  const isNewRoom = useCallback(() => roomId === undefined, [roomId]);
```

Comment:
Do we need a function/callback for this? Wouldn't just `const isNewRoom = roomId === undefined;` work?

Reply:
We don't, actually, I'm not even sure how that ended up being a callback.